ql/pricingengines/credit/integralcdsengine.cpp

```diff
@@ -52,20 +52,39 @@ namespace QuantLib {
 
         // Upfront Flow NPV. Either we are on-the-run (no flow)
         // or we are forward start
+
+        // date determining the probability survival so we have to pay
+        //   the upfront flows (did not knock out)
+        Date effectiveProtectionStart =
+            arguments_.protectionStart > probability_->referenceDate() ?
+                arguments_.protectionStart : probability_->referenceDate();
+
+        Probability nonKnockOut = 
+            probability_->survivalProbability(effectiveProtectionStart);
+
         Real upfPVO1 = 0.0;
-        if(!arguments_.upfrontPayment->hasOccurred(
+        results_.upfrontNPV = 0.0;
+        if (arguments_.upfrontPayment && 
```

Comment:
You could check for `upfront->amount() == 0` and skip the calculation altogether in that case (see also my other comment on initializing the upfront).